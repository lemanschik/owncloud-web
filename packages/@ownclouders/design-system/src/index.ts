import { getSizeClass } from './utils/sizeClasses'
import './utils/webFontLoader'

import * as components from './components'
import * as directives from './directives'

const initializeCustomProps = (tokens = [], prefix) => {
  for (const param in tokens) {
    initializeCustomProp(prefix + param, tokens[param])
  }
}

const initializeCustomProp = (key: string, value: string | undefined) => {
  if (value === undefined) {
    return
  }
  ;(document.querySelector(':root') as HTMLElement).style.setProperty('--oc-' + key, value)
}

export default {
  install(app, options: any = {}) {
    const themeOptions = options.tokens
    initializeCustomProps(themeOptions?.breakpoints, 'breakpoint-')
    initializeCustomProps(themeOptions?.colorPalette, 'color-')
    initializeCustomProps(themeOptions?.fontSizes, 'font-size-')
    initializeCustomProps(themeOptions?.sizes, 'size-')
    initializeCustomProps(themeOptions?.spacing, 'space-')
    initializeCustomProp('font-family', themeOptions?.fontFamily)

    Object.values(components).forEach((c) => app.component(c.name, c))
    Object.values(directives).forEach((d) => app.directive(d.name, d))
  }
}

export const utils = { getSizeClass }
export * as components from './components'
export * as composables from './composables'
export * as directives from './directives'
export * as helpers from './helpers'
