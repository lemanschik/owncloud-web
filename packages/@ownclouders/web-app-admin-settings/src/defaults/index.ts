export const supportedLogoMimeTypes = ['image/gif', 'image/jpeg', 'image/png']
