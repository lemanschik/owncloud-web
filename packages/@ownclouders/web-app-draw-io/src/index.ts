import translations from '../l10n/translations.json'
import App from './App.vue'

const routes = [
  {
    name: 'draw-io',
    path: '/:driveAliasAndItem(.*)?',
    component: App,
    meta: {
      authContext: 'hybrid',
      patchCleanPath: true
    }
  }
]

const appInfo = {
  name: 'Draw.io',
  id: 'draw-io',
  icon: 'grid',
  extensions: [
    {
      extension: 'drawio',
      routeName: 'draw-io',
      newFileMenu: {
        menuTitle($gettext) {
          return $gettext('Draw.io document')
        }
      }
    },
    {
      extension: 'vsdx',
      routeName: 'draw-io'
    }
  ]
}

export default {
  appInfo,
  routes,
  translations
}
