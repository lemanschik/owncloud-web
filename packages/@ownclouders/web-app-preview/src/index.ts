import translations from '../l10n/translations.json'
import * as app from './App.vue'
const { default: App, mimeTypes, appId } = app as any

// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const routes = [
  {
    path: '/:driveAliasAndItem(.*)?',
    component: App,
    name: 'media',
    meta: {
      authContext: 'hybrid',
      title: $gettext('Preview'),
      patchCleanPath: true
    }
  }
]

const routeName = 'preview-media'

const appInfo = {
  name: $gettext('Preview'),
  id: appId,
  icon: 'eye',
  extensions: mimeTypes().map((mimeType) => ({
    canBeDefault: true,
    mimeType,
    routeName,
    label: $gettext('Preview')
  }))
}

export default {
  appInfo,
  routes,
  translations,
  mimeTypes
}
