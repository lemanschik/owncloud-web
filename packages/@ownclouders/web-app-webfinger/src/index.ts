import translations from '../l10n/translations.json'
import Resolve from './views/Resolve.vue'

// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Webfinger'),
  id: 'webfinger',
  icon: 'fingerprint',
  isFileEditor: false
}

const routes = () => [
  {
    name: 'webfinger-root',
    path: '/',
    redirect: () => {
      return { name: 'webfinger-resolve' }
    },
    meta: {
      authContext: 'anonymous'
    }
  },
  {
    path: '/resolve',
    name: 'webfinger-resolve',
    component: Resolve,
    meta: {
      authContext: 'idp',
      title: $gettext('Resolve destination'),
      entryPoint: true
    }
  }
]

export default {
  appInfo,
  routes,
  translations
}
