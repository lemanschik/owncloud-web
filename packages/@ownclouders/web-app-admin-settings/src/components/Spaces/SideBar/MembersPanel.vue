<template>
  <div class="oc-ml-s">
    <oc-text-input
      v-model="filterTerm"
      class="oc-text-truncate oc-mr-s oc-mt-m"
      :label="$gettext('Filter members')"
    />
    <div ref="membersListRef" data-testid="space-members">
      <div v-if="!filteredSpaceMembers.length">
        <h3 class="oc-text-bold oc-text-medium" v-text="$gettext('No members found')" />
      </div>
      <div
        v-if="filteredSpaceManagers.length"
        class="oc-mb-m"
        data-testid="space-members-role-managers"
      >
        <h3 class="oc-text-bold oc-text-medium" v-text="$gettext('Managers')" />
        <members-role-section :members="filteredSpaceManagers" />
      </div>
      <div
        v-if="filteredSpaceEditors.length"
        class="oc-mb-m"
        data-testid="space-members-role-editors"
      >
        <h3 class="oc-text-bold oc-text-medium" v-text="$gettext('Editors')" />
        <members-role-section :members="filteredSpaceEditors" />
      </div>
      <div
        v-if="filteredSpaceViewers.length"
        class="oc-mb-m"
        data-testid="space-members-role-viewers"
      >
        <h3 class="oc-text-bold oc-text-medium" v-text="$gettext('Viewers')" />
        <members-role-section :members="filteredSpaceViewers" />
      </div>
    </div>
  </div>
</template>
<script lang="ts">
import { computed, defineComponent, inject, ref, watch, unref } from 'vue'
import { Resource } from 'web-client/src'
import MembersRoleSection from './MembersRoleSection.vue'
import Fuse from 'fuse.js'
import Mark from 'mark.js'
import { spaceRoleEditor, spaceRoleManager, spaceRoleViewer } from 'web-client/src/helpers/share'
import { defaultFuseOptions } from 'web-pkg/src/helpers'

export default defineComponent({
  name: 'MembersPanel',
  components: { MembersRoleSection },
  setup() {
    const resource = inject<Resource>('resource')
    const filterTerm = ref('')
    const markInstance = ref(null)
    const membersListRef = ref(null)
    const filterMembers = (collection, term) => {
      if (!(term || '').trim()) {
        return collection
      }

      const searchEngine = new Fuse(collection, { ...defaultFuseOptions, keys: ['displayName'] })
      return searchEngine.search(term).map((r) => r.item)
    }

    const spaceMembers = computed(() => {
      return [
        ...unref(resource).spaceRoles.manager.map((r) => ({
          ...r,
          roleType: spaceRoleManager.name
        })),
        ...unref(resource).spaceRoles.editor.map((r) => ({
          ...r,
          roleType: spaceRoleEditor.name
        })),
        ...unref(resource).spaceRoles.viewer.map((r) => ({
          ...r,
          roleType: spaceRoleViewer.name
        }))
      ].sort((a, b) => a.displayName.localeCompare(b.displayName))
    })

    const filteredSpaceMembers = computed(() => {
      return filterMembers(unref(spaceMembers), unref(filterTerm))
    })
    const filteredSpaceManagers = computed(() => {
      return unref(filteredSpaceMembers).filter((m) => m.roleType === spaceRoleManager.name)
    })
    const filteredSpaceEditors = computed(() => {
      return unref(filteredSpaceMembers).filter((m) => m.roleType === spaceRoleEditor.name)
    })
    const filteredSpaceViewers = computed(() => {
      return unref(filteredSpaceMembers).filter((m) => m.roleType === spaceRoleViewer.name)
    })

    watch(filterTerm, () => {
      if (unref(membersListRef)) {
        markInstance.value = new Mark(unref(membersListRef))
        unref(markInstance).unmark()
        unref(markInstance).mark(unref(filterTerm), {
          element: 'span',
          className: 'highlight-mark'
        })
      }
    })

    return {
      filterTerm,
      filteredSpaceMembers,
      filteredSpaceManagers,
      filteredSpaceEditors,
      filteredSpaceViewers,
      membersListRef
    }
  }
})
</script>
