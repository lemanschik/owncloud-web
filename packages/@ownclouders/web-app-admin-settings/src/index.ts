import translations from '../l10n/translations.json'
import General from './views/General.vue'
import Users from './views/Users.vue'
import Groups from './views/Groups.vue'
import Spaces from './views/Spaces.vue'
import { Ability } from 'web-pkg'
import { AppNavigationItem } from 'web-pkg/src/apps'

// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Administration Settings'),
  id: 'admin-settings',
  icon: 'settings-4',
  isFileEditor: false
}

const routes = ({ $ability }: { $ability: Ability }) => [
  {
    path: '/',
    redirect: () => {
      if ($ability.can('read-all', 'Setting')) {
        return { name: 'admin-settings-general' }
      }
      if ($ability.can('read-all', 'Account')) {
        return { name: 'admin-settings-users' }
      }
      if ($ability.can('read-all', 'Group')) {
        return { name: 'admin-settings-groups' }
      }
      if ($ability.can('read-all', 'Space')) {
        return { name: 'admin-settings-spaces' }
      }
      throw Error('Insufficient permissions')
    }
  },
  {
    path: '/general',
    name: 'admin-settings-general',
    component: General,
    beforeEnter: (to, from, next) => {
      if (!$ability.can('read-all', 'Setting')) {
        next({ path: '/' })
      }
      next()
    },
    meta: {
      authContext: 'user',
      title: $gettext('General')
    }
  },
  {
    path: '/users',
    name: 'admin-settings-users',
    component: Users,
    beforeEnter: (to, from, next) => {
      if (!$ability.can('read-all', 'Account')) {
        next({ path: '/' })
      }
      next()
    },
    meta: {
      authContext: 'user',
      title: $gettext('Users')
    }
  },
  {
    path: '/groups',
    name: 'admin-settings-groups',
    component: Groups,
    beforeEnter: (to, from, next) => {
      if (!$ability.can('read-all', 'Group')) {
        next({ path: '/' })
      }
      next()
    },
    meta: {
      authContext: 'user',
      title: $gettext('Groups')
    }
  },
  {
    path: '/spaces',
    name: 'admin-settings-spaces',
    component: Spaces,
    beforeEnter: (to, from, next) => {
      if (!$ability.can('read-all', 'Space')) {
        next({ path: '/' })
      }
      next()
    },
    meta: {
      authContext: 'user',
      title: $gettext('Spaces')
    }
  }
]

const navItems = ({ $ability }: { $ability: Ability }): AppNavigationItem[] => [
  {
    name: $gettext('General'),
    icon: 'settings-4',
    route: {
      path: `/${appInfo.id}/general?`
    },
    enabled: () => {
      return $ability.can('read-all', 'Setting')
    }
  },
  {
    name: $gettext('Users'),
    icon: 'user',
    route: {
      path: `/${appInfo.id}/users?`
    },
    enabled: () => {
      return $ability.can('read-all', 'Account')
    }
  },
  {
    name: $gettext('Groups'),
    icon: 'group-2',
    route: {
      path: `/${appInfo.id}/groups?`
    },
    enabled: () => {
      return $ability.can('read-all', 'Group')
    }
  },
  {
    name: $gettext('Spaces'),
    icon: 'layout-grid',
    route: {
      path: `/${appInfo.id}/spaces?`
    },
    enabled: () => {
      return $ability.can('read-all', 'Space')
    }
  }
]

export default {
  appInfo,
  routes,
  translations,
  navItems
}
