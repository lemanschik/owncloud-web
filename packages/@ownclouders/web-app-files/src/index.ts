import App from './App.vue'
import Favorites from './views/Favorites.vue'
import FilesDrop from './views/FilesDrop.vue'
import SharedWithMe from './views/shares/SharedWithMe.vue'
import SharedWithOthers from './views/shares/SharedWithOthers.vue'
import SharedViaLink from './views/shares/SharedViaLink.vue'
import SpaceDriveResolver from './views/spaces/DriveResolver.vue'
import SpaceProjects from './views/spaces/Projects.vue'
import TrashOverview from './views/trash/Overview.vue'
import translations from '../l10n/translations.json'
import quickActions from './quickActions'
import store from './store'
import { SDKSearch } from './search'
import { eventBus } from 'web-pkg/src/services/eventBus'
import { Registry, ArchiverService } from './services'
import fileSideBars from './fileSideBars'
import { buildRoutes } from './router'
import get from 'lodash-es/get'
import { AppNavigationItem, AppReadyHookArgs } from 'web-pkg/src/apps'

// dirty: importing view from other extension within project
import SearchResults from '../../web-app-search/src/views/List.vue'
import { isPersonalSpaceResource } from 'web-client/src/helpers'

// just a dummy function to trick gettext tools
function $gettext(msg) {
  return msg
}

const appInfo = {
  name: $gettext('Files'),
  id: 'files',
  icon: 'resource-type-folder',
  isFileEditor: false,
  extensions: [],
  fileSideBars
}
const navItems = (context): AppNavigationItem[] => {
  return [
    {
      name(capabilities) {
        return capabilities.spaces?.enabled ? $gettext('Personal') : $gettext('All files')
      },
      icon: appInfo.icon,
      route: {
        path: `/${appInfo.id}/spaces/personal`
      },
      enabled(capabilities) {
        if (!capabilities.spaces?.enabled) {
          return true
        }
        return !!context?.$store?.getters['runtime/spaces/spaces'].find((drive) =>
          isPersonalSpaceResource(drive)
        )
      }
    },
    {
      name: $gettext('Favorites'),
      icon: 'star',
      route: {
        path: `/${appInfo.id}/favorites`
      },
      enabled(capabilities) {
        return capabilities.files?.favorites
      }
    },
    {
      name: $gettext('Shares'),
      icon: 'share-forward',
      route: {
        path: `/${appInfo.id}/shares`
      },
      activeFor: [{ path: `/${appInfo.id}/spaces/share` }],
      enabled(capabilities) {
        return capabilities.files_sharing?.api_enabled !== false
      }
    },
    {
      name: $gettext('Spaces'),
      icon: 'layout-grid',
      route: {
        path: `/${appInfo.id}/spaces/projects`
      },
      activeFor: [{ path: `/${appInfo.id}/spaces/project` }],
      enabled(capabilities) {
        return capabilities.spaces?.projects
      }
    },
    {
      name: $gettext('Deleted files'),
      icon: 'delete-bin-5',
      route: {
        path: `/${appInfo.id}/trash/overview`
      },
      activeFor: [{ path: `/${appInfo.id}/trash` }],
      enabled(capabilities) {
        return capabilities.dav?.trashbin === '1.0'
      }
    }
  ]
}

export default {
  appInfo,
  store,
  routes: buildRoutes({
    App,
    Favorites,
    FilesDrop,
    SearchResults,
    Shares: {
      SharedViaLink,
      SharedWithMe,
      SharedWithOthers
    },
    Spaces: {
      DriveResolver: SpaceDriveResolver,
      Projects: SpaceProjects
    },
    Trash: {
      Overview: TrashOverview
    }
  }),
  navItems,
  quickActions,
  translations,
  ready({ router, store, globalProperties }: AppReadyHookArgs) {
    const { $clientService } = globalProperties
    Registry.sdkSearch = new SDKSearch(store, router, $clientService)

    // when discussing the boot process of applications we need to implement a
    // registry that does not rely on call order, aka first register "on" and only after emit.
    eventBus.publish('app.search.register.provider', Registry.sdkSearch)
    globalProperties.$archiverService = new ArchiverService(
      $clientService,
      store.getters.configuration.server || window.location.origin,
      get(store, 'getters.capabilities.files.archivers', [
        {
          enabled: true,
          version: '1.0.0',
          formats: ['tar', 'zip'],
          archiver_url: `${store.getters.configuration.server}index.php/apps/files/ajax/download.php`
        }
      ])
    )
  }
}
