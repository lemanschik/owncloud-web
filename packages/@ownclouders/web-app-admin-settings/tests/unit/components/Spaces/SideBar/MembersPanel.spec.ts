import MembersPanel from '../../../../../src/components/Spaces/SideBar/MembersPanel.vue'
import { defaultPlugins, shallowMount } from 'web-test-helpers'
import { mock } from 'jest-mock-extended'
import { SpaceResource } from 'web-client/src/helpers'

const spaceMock = mock<SpaceResource>({
  spaceRoles: {
    manager: [{ kind: 'user', displayName: 'admin' }],
    editor: [
      { kind: 'user', displayName: 'einstein' },
      { kind: 'group', displayName: 'physic-haters' }
    ],
    viewer: [{ kind: 'user', displayName: 'marie' }]
  }
})

const selectors = {
  membersRolePanelStub: 'members-role-section-stub'
}

describe('MembersPanel', () => {
  it('should render all members accordingly to their role assignments', () => {
    const { wrapper } = getWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should filter members accordingly to the entered search term', async () => {
    const userToFilterFor = spaceMock.spaceRoles.editor[0]
    const { wrapper } = getWrapper()
    wrapper.vm.filterTerm = 'ein'
    await wrapper.vm.$nextTick
    expect(wrapper.findAll(selectors.membersRolePanelStub).length).toBe(1)
    expect(
      wrapper.findComponent<any>(selectors.membersRolePanelStub).props().members[0].displayName
    ).toEqual(userToFilterFor.displayName)
  })
  it('should display an empty result if no matching members found', async () => {
    const { wrapper } = getWrapper()
    wrapper.vm.filterTerm = 'no-match'
    await wrapper.vm.$nextTick
    expect(wrapper.findAll(selectors.membersRolePanelStub).length).toBe(0)
    expect(wrapper.html()).toMatchSnapshot()
  })
})

function getWrapper({ spaceResource = spaceMock } = {}) {
  return {
    wrapper: shallowMount(MembersPanel, {
      global: {
        plugins: [...defaultPlugins()],
        provide: { resource: spaceResource }
      }
    })
  }
}
