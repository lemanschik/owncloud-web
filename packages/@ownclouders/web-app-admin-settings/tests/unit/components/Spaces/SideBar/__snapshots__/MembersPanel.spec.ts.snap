// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MembersPanel should display an empty result if no matching members found 1`] = `
<div class="oc-ml-s">
  <oc-text-input-stub class="oc-text-truncate oc-mr-s oc-mt-m" clearbuttonaccessiblelabel="" clearbuttonenabled="false" disabled="false" fixmessageline="false" id="oc-textinput-3" label="Filter members" modelvalue="no-match" readonly="false" type="text"></oc-text-input-stub>
  <div>
    <div>
      <h3 class="oc-text-bold oc-text-medium">No members found</h3>
    </div>
    <!--v-if-->
    <!--v-if-->
    <!--v-if-->
  </div>
</div>
`;

exports[`MembersPanel should render all members accordingly to their role assignments 1`] = `
<div class="oc-ml-s">
  <oc-text-input-stub class="oc-text-truncate oc-mr-s oc-mt-m" clearbuttonaccessiblelabel="" clearbuttonenabled="false" disabled="false" fixmessageline="false" id="oc-textinput-1" label="Filter members" modelvalue="" readonly="false" type="text"></oc-text-input-stub>
  <div>
    <!--v-if-->
    <div class="oc-mb-m">
      <h3 class="oc-text-bold oc-text-medium">Managers</h3>
      <members-role-section-stub members="[object Object]"></members-role-section-stub>
    </div>
    <div class="oc-mb-m">
      <h3 class="oc-text-bold oc-text-medium">Editors</h3>
      <members-role-section-stub members="[object Object],[object Object]"></members-role-section-stub>
    </div>
    <div class="oc-mb-m">
      <h3 class="oc-text-bold oc-text-medium">Viewers</h3>
      <members-role-section-stub members="[object Object]"></members-role-section-stub>
    </div>
  </div>
</div>
`;
